ibis/expr/types/relations.py

```diff
@@ -4672,13 +4650,19 @@ def value_counts(self) -> ir.Table:
 @public
 class CachedTable(Table):
     def __exit__(self, *_):
-        self.release()
+        self.release(stacklevel=3)
 
     def __enter__(self):
         return self
 
-    def release(self):
+    def release(self, stacklevel=2):
         """Release the underlying expression from the cache."""
+        warnings.warn(
```

Comment:
Should we remove these in version 10? If so, let's add a message here about that.

Alternatively, I think `util.deprecated` might be possible to use here, which would give a slightly more standardized (standard from the perspective of Ibis anyway) error message.

Reply:
I actually would prefer we keep these ways to explicitly release cached tables. Complex code can make understanding if the cached tables has any references left tricky, and can also result in code littered in dels (which is a code smell imo). There should be a way to make both patterns still work after this change.

Reply:
Yeah, that makes sense to me. Keeping these seems like a good idea.

Reply:
> There should be a way to make both patterns still work after this change.

Both patterns do work with the current implementation and can continue to work (the linear search in `release` should maybe be helped by an inverse dictionary).

Reply:
Do we want to remove the `DeprecationWarning` and maybe update the docs and release notes to make the new behaviour better known, then?

Reply:
Yeah, that sounds like a good idea!

Reply:
Ok, reverted the changes around the `DeprecationWarning` and added some additional detail to the docstring.